src/odin/maneuversbuilder.cc

```diff
@@ -2369,6 +2397,76 @@ bool ManeuversBuilder::AreRampManeuversCombinable(std::list<Maneuver>::iterator
   return false;
 }
 
+bool ManeuversBuilder::IsNextManeuverObvious(std::list<Maneuver>& maneuvers,
+                                             std::list<Maneuver>::iterator curr_man,
+                                             std::list<Maneuver>::iterator next_man) const {
+  // The next maneuver must be a continue maneuver
+  if ((next_man->type() == DirectionsLeg_Maneuver_Type_kContinue)) {
+    // Get the node between the the current and next maneuver
+    auto node = trip_path_->GetEnhancedNode(next_man->begin_node_index());
+
+    // Return true if there are no traversable intersecting edges
+    if (node && !node->HasTraversableIntersectingEdge(next_man->travel_mode())) {
+      return true;
+    }
+
+    // Return false if the maneuver has an exit number
+    if (next_man->HasExitNumberSign()) {
+      return false;
+    }
+
+    // Process ramp forks
+    if (curr_man->ramp() && curr_man->fork() && !curr_man->contains_obvious_maneuver()) {
+      // Obvious if Keep straight and continue
+      if (curr_man->type() == DirectionsLeg_Maneuver_Type_kStayStraight) {
+        return true;
+      } else {
+        if (node) {
+          IntersectingEdgeCounts xedge_counts;
+          node->CalculateRightLeftIntersectingEdgeCounts(curr_man->end_heading(),
+                                                         curr_man->travel_mode(), xedge_counts);
+
+          // TODO: we could enhance in the future
+          // Obvious if left fork and no left edges at intersection
+          if ((curr_man->type() == DirectionsLeg_Maneuver_Type_kStayLeft) &&
+              (xedge_counts.left == 0)) {
+            return true;
+          }
+
+          // Obvious if right fork and no right edges at intersection
+          if ((curr_man->type() == DirectionsLeg_Maneuver_Type_kStayRight) &&
+              (xedge_counts.right == 0)) {
+            return true;
+          }
+        }
+      }
+      return false;
+    }
+
+    // Return true if a short continue maneuver
+    // and the following maneuver is not a continue
+    if (next_man->length(Options_Units_kilometers) < kShortContinueThreshold) {
```

Comment:
Could this function be called with `next_man` pointing to the last/arrival maneuver? If so, should we return `false` if `next_next_man == maneuvers.end()` ?

Reply:
`next_man` must be a `DirectionsLeg_Maneuver_Type_kContinue`